consensus/ibft/snapshot_test.go

```diff
@@ -50,36 +52,42 @@ func (t *testerAccount) Address() types.Address {
 	return crypto.PubKeyToAddress(&t.priv.PublicKey)
 }
 
-func (t *testerAccount) sign(h *types.Header) *types.Header {
-	h, _ = writeSeal(t.priv, h)
+func (t *testerAccount) sign(h *types.Header) (*types.Header, error) {
+	signer := signer.NewECDSASignerFromKey(t.priv)
 
-	return h
+	return signer.WriteSeal(h)
 }
 
 type testerAccountPool struct {
+	t        *testing.T
```

Comment:
I honestly don't think we should be dragging a reference to `testing.T` around in the helper methods.

Why not just return errors for methods that can error out?